flink-libraries/flink-table/src/test/scala/org/apache/flink/table/runtime/dataset/DataSetUserDefinedFunctionITCase.scala

```diff
@@ -117,6 +117,187 @@ class DataSetUserDefinedFunctionITCase(
   }
 
   @Test
+  def testDynamicSchema(): Unit = {
```

Comment:
Also, these are many integration tests which all run a full job. 
Aren't `TableTestBase` sufficient to check if the plan is correctly translated? IMO, we should have only few integration tests and more unit tests based on `TableTestBase`.

What do you think?